### PR TITLE
Enable SyncBatchNorm in CPU backend

### PR DESCRIPTION
**Pull request by yanbing-j**
SyncBatchNorm is a feature in Pytorch that is currently supported only for GPU. It allows for calculation of batch statistics for Batch norm over multiple ranks during distributed training. By using SyncBatchNorm, training workload can scale-out over many ranks  without necessarily increasing global batch size and thereby epochs to convergence.

This PR is to enable SyncBatchNorm in CPU backend, to help Resnet50 training in CPU.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: torch/testing/_internal/distributed/distributed_test.py [hunk 1]
```diff
@@ -5009,30 +5013,36 @@ def test_DistributedDataParallel_SyncBatchNorm_Channels_Last(self):
...
-                model_gpu, device_ids=[rank]
+                model_base, device_ids=device_ids
             )
 
             memory_format = torch.channels_last
-            input_gpu = (
+            input_base = (
                 torch.randn(global_bs, 2, 4, 4, dtype=torch.float)
-                .cuda(rank)
+                .to(device)
```
Comment (vadimkantorov) [inline torch/testing/_internal/distributed/distributed_test.py:5030]: i wonder if to(device), to(memory_format) could be integrated directly as arguments to randn(...). It seems that device = device definitely could be moved there, but memory_format seems not supported by randn(...)

Comment (yanbing-j) [reply torch/testing/_internal/distributed/distributed_test.py:5030]: Indeed, `to(device)` could be integrated in `randn`, but `memory_format` could not.

`@overload`
`def randn(size: _size, *, generator: Optional[Generator], names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(*size: _int, generator: Optional[Generator], names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(size: _size, *, generator: Optional[Generator], out: Optional[Tensor]=None, dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(*size: _int, generator: Optional[Generator], out: Optional[Tensor]=None, dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(size: _size, *, out: Optional[Tensor]=None, dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(*size: _int, out: Optional[Tensor]=None, dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`
`@overload`
`def randn(size: _size, *, names: Optional[Sequence[Union[str, ellipsis, None]]], dtype: _dtype=None, layout: Optional[_layout]=strided, device: Union[_device, str, None]=None, pin_memory: _bool=False, requires_grad: _bool=False) -> Tensor: ...`

**Comment by yanbing-j** [comment]
Hi @mrshenli @awgu , could you please help review this PR? Thank you.

**Comment by yanbing-j** [comment]
Hi @vadimkantorov @mrshenli @H-Huang @awgu @bdhirsh @jbschlosser, could you please help review this PR? Thank you!
